packages/ui/src/components/TagList.tsx

```diff
@@ -0,0 +1,163 @@
+      const otherExpandedTags = prevExpanded.filter((name) => name !== tagName);
+      return isExpanding ? [...otherExpandedTags, tagName] : otherExpandedTags;
+    });
+  };
+
+  const isEditingTag = (tagName: string) => editingTagName === tagName;
+
+  const handleDescriptionClick = (tagName: string, description: string) => {
+    setEditingTagName(tagName);
+    setCurrentCode(description);
+  };
+
+  const updateDescription = (tagName: string) => {
+    const updatedTags = updatedTagList.map((tag) => {
+      if (tag.name === tagName) {
+        return { ...tag, description: currentCode };
+      }
+      return tag;
+    });
+    setUpdatedTagList(updatedTags);
+    setEditingTagName(null); 
+  };
+
+  const isTagExpanded = (tagName: string) => expandedTagNames.includes(tagName);
+
+  const defaultActions = (): IAction[] => [
+    {
+      title: (
+        <>
+          <PencilAltIcon />
+          &nbsp;Rename
+        </>
+      ),
+      onClick: () => { },
+    },
+    {
+      title: (
+        <>
+          <TrashIcon />
+          &nbsp;Delete
+        </>
+      ),
+      onClick: () => { },
+    },
+  ];
+
+  const editCustomControl = (
+    <CodeEditorControl
+      icon={<CheckIcon color="blue" />}
+      aria-label="Save code"
+      tooltipProps={{ content: 'Save' }}
+      onClick={() => editingTagName && updateDescription(editingTagName)} 
+    />
+  );
+
+  const cancelCustomControl = (
+    <CodeEditorControl
+      icon={<TimesIcon />}
+      aria-label="Cancel edit"
+      tooltipProps={{ content: 'Cancel edit' }}
+      onClick={() => setEditingTagName(null)} 
+    />
+  );
+
+  return (
+    <Table aria-label="Tag List">
+      {updatedTagList.map((tag, rowIndex) => (
+        <Tbody key={tag.name} isExpanded={isTagExpanded(tag.name)}>
+          <Tr>
+            <Td
+              expand={{
+                rowIndex,
+                isExpanded: isTagExpanded(tag.name),
+                onToggle: () =>
+                  setTagExpanded(tag.name, !isTagExpanded(tag.name)),
+                expandId: 'tag-expandable',
+              }}
+            />
+            <Td dataLabel="Name">
+              <>
+                <TagIcon /> {tag.name}
+              </>
+            </Td>
+            <Td dataLabel="Tag Description">{tag.description}</Td>
+            <Td isActionCell>
+              <ActionsColumn items={defaultActions()} />
+            </Td>
+          </Tr>
+          {isTagExpanded(tag.name) && (
+            <Tr isExpanded={isTagExpanded(tag.name)}>
+              <Td colSpan={2}>
+                <ExpandableRowContent>
+                  <DescriptionList>
+                    <DescriptionListGroup>
+                      <DescriptionListTerm>Description</DescriptionListTerm>
+                      <DescriptionListDescription
+                        onClick={() =>
+                          handleDescriptionClick(tag.name, tag.description)
+                        }
```

Comment:
This is just my opinion, but I think clicking on the description to edit it, is not clear enough since there's no hint about this functionality.

If we compare it with how renaming the tag works, we have a mechanism that states the rename option.

@EricWittmann, I think this behavior comes from the existing UI, but could something like this work? Please take a look at the pencil icon that controls whether you are in edit mode or not, maybe instead of having to click on the description itself, we could show the pencil icon and that will trigger the edition.

Please let me know your thoughts.

![image](https://github.com/user-attachments/assets/6269a7fa-cdc9-45e6-8435-875056936635)

Reply:
I think generally we should **not** assume that the existing editor's user interactions are always the right approach.  If there is a current patternfly standard, I think we should try to align with that wherever possible.

In this particular case, I'd be very happy with a pencil icon if that's what you guys think works best!

Reply:
ok, thanks for clarifying that.

@hemahg, what do you think about moving forward with this PR and then incorporating the InlineEditor and replacing this behavior? We've implemented it already in Kaoto

Reply:
Sure, I'll make the changes accordingly.